carbonmark/components/pages/Users/SellerConnected/Forms/EditListing.tsx

```diff
@@ -119,6 +120,12 @@ export const EditListing: FC<Props> = (props) => {
                   value: Number(totalAvailableQuantity),
                   message: t`Available balance exceeded`,
                 },
+                validate: {
+                  isWholeNumber: (value) =>
+                    !props.listing.project.id.startsWith("ICR") ||
+                    Number.isInteger(parseFloat(value)) ||
+                    t`ICR credits can only be listed in whole tonnes`,
```

Comment:
I've seen this in a few places now, can you please extract it as it's own validator.